docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,38 +1,96 @@
+* cloud authentication/authorization: your cloud may require a token or IAM rules to allow access to their APIs
+* kubernetes authentication/authorization: cloud-controller-manager may need RBAC rules set to speak to the kubernetes apiserver
+* high availabilty: like kube-controller-manager, you may want a high available setup for cloud controller manager using leader election (on by default).
 
 ### Running cloud-controller-manager
 
-To run `cloud-controller-manager`, add it to your existing Kubernetes cluster as a Master component. All other master components except `kube-controller-manager` can be run without any changes.
+Successfully running cloud-controller-manager requires some changes to your cluster configuration.
+
+* `kube-apiserver` and `kube-controller-manager` must not specify a `--cloud-provider` flag or must set it to `--cloud-provider=external`. This ensures that it does not run any cloud specific loops that would be run by cloud controller manager. In the future, this flag will be deprecated and removed.
+* `kubelet` must run with `--cloud-provider=external`. This is to ensure that the kubelet is aware that it must be initialized by the cloud controller manager before it is scheduled any work.
+
+Keep in mind that setting up your cluster to use cloud controller manager will change your cluster behaviour in a few ways:
+
+* kubelets specifying `--cloud-provider=external` will add a taint `node.cloudprovider.kubernetes.io/uninitialized` with an effect `NoSchedule` during initialization. This marks the node as needing a second initialization from an external controller before it can be scheduled work. Note that in the event that cloud controller manager is not available, new nodes in the cluster will be left unscheduable. The taint is important since the scheduler may require cloud specific information about nodes such as it's region or type (high cpu, gpu, high memory, spot instance, etc).
+* cloud information about nodes in the cluster will no longer be retrieved using local metadata, but instead all API calls to retreive node information will go through cloud controller manager. This may mean you can restrict access to your cloud API on the kubelets for better security. For larger clusters you may want to consider if cloud controller manager will hit rate limits since it is now responsible for almost all API calls to your cloud from within the cluster.
+* kube-apiserver should not run the Persistent Volume Label admission controller either since the cloud-controller-manager takes over labeling persistent volumes. To prevent the Persistent Volume Label admission plugin from running, make sure the kube-apiserver has a --admission-control flag with a value that does not include PersistentVolumeLabel.
```

Comment:
You also need to create an InitializerConfiguration so that the Persistent Volume Label controller will actually act on new persistent volumes.  See https://github.com/kubernetes/kubernetes.github.io/blob/release-1.8/docs/tasks/administer-cluster/running-cloud-controller.md